src/rewriter/mkrewrites.py

```diff
@@ -96,10 +37,23 @@ def gen_mk_skolem(name, sort):
         sort_code = 'nm->realType()'
     elif sort.base == BaseSort.String:
         sort_code = 'nm->stringType()'
-    elif sort.base == BaseSort.String:
-        sort_code = 'nm->stringType()'
     elif sort.base == BaseSort.RegLan:
         sort_code = 'nm->regExpType()'
+    elif sort.base == BaseSort.String:
+        sort_code = 'nm->stringType()'
+    elif sort.base == BaseSort.AbsArray:
+        sort_code = 'nm->mkAbstractType(Kind::ARRAY_TYPE)'
+    elif sort.base == BaseSort.AbsBitVec:
+        sort_code = 'nm->mkAbstractType(Kind::BITVECTOR_TYPE)'
+    elif sort.base == BaseSort.AbsSeq:
+        sort_code = 'nm->mkAbstractType(Kind::SEQUENCE_TYPE)'
+    elif sort.base == BaseSort.AbsSet:
+        sort_code = 'nm->mkAbstractType(Kind::SET_TYPE)'
+    elif sort.base == BaseSort.AbsAbs:
+        sort_code = 'nm->mkAbstractType(Kind::ABSTRACT_TYPE)'
+    elif sort.base == BaseSort.BitVec:
+        # This will result in a compilation error for variable BitVec sizes.
```

Comment:
Shouldn't this be guarded in this script (if not supported) instead of letting the compiler fail?

Reply:
fixed